.github/workflows/e2e_libvirt.yaml

```diff
@@ -81,35 +97,58 @@ jobs:
            docker system prune -a -f
         working-directory: src/cloud-api-adaptor/podvm
 
-      - name: Get the install directory
-        if: ${{ inputs.install_directory_artifact != '' }}
-        uses: actions/download-artifact@v4
-        with:
-          name: ${{ inputs.install_directory_artifact }}
-          path: src/cloud-api-adaptor/install
+      - name: Use oras to get qcow2 from ${{ inputs.podvm_image }}
+        if: ${{ inputs.oras }}
+        run: |
+          oras pull ${{ inputs.podvm_image }}
+          tar xvJpf podvm.tar.xz
+          qcow2=$(find ./*.qcow2)
+          echo "PODVM_QCOW2=$(pwd)/${qcow2}" >> "$GITHUB_ENV"
+        working-directory: src/cloud-api-adaptor/podvm
 
       - name: Config Libvirt
         run: |
           export TEST_E2E_SECURE_COMMS="${{ inputs.secure_comms }}"
           ./libvirt/config_libvirt.sh
-          echo "CAA_IMAGE=\"${{ inputs.caa_image }}\"" >> libvirt.properties
           # For debugging
           cat libvirt.properties
 
-      - uses: oras-project/setup-oras@v1
-        with:
-          version: ${{ env.ORAS_VERSION }}
-
       - name: Install gh cli
         run: |
           sudo apt install -y gh
 
+      - name: Double check that OVMF is installed
+        run: |
+          sudo apt install -y ovmf
```

Comment:
I think it makes sense to always perform, `sudo apt-get install update` before installing. in theory you only need to do it once, but when rearranging steps that can lead to breakage sometimes.

Reply:
Good idea. I've updated this and added to a couple of the other places it was missing too in this commit.